base/array.jl

```diff
@@ -1376,10 +1375,9 @@ function prepend!(a::Vector{T}, items::Union{AbstractVector{<:T},Tuple}) where T
     return a
 end
 
-prepend!(a::Vector, iter) = _prepend!(a, IteratorSize(iter), iter)
-pushfirst!(a::Vector, iter...) = prepend!(a, iter)
-
-prepend!(a::AbstractVector, iter...) = foldr((v, a) -> prepend!(a, v), iter, init=a)
+prepend!(a::AbstractVector, iter) = _prepend!(a, IteratorSize(iter), iter)
+pushfirst!(a::AbstractVector, iter...) = prepend!(a, iter)
+prepend!(a::AbstractVector, iter...) = (for i = length(iter):-1:1; prepend!(a, iter[i]); end; return a)
```

Comment:
alternatively? I guess these should generate the same code though?
```suggestion
prepend!(a::AbstractVector, iter...) = (for i = reverse(iter); prepend!(a, i); end; return a)
```